pkg/tfbridge/validate_input_types.go

```diff
@@ -51,6 +56,12 @@ func (v *PulumiInputValidator) validatePropertyMap(
 		objectValue := propertyMap[objectKey]
 		objType, knownType := propertyTypes[string(objectKey)]
 		if !knownType {
+			if v.validateUnknownTypes {
+				return []TypeFailure{{
+					Reason:       fmt.Sprintf("an unexpected argument %q was provided", string(objectKey)),
+					ResourcePath: propertyPath.String(),
+				}}
```

Comment:
We should present as many errors as possible, not just the first one that comes up.
```suggestion
				failures = append(failures, TypeFailure{
					Reason:       fmt.Sprintf("an unexpected argument %q was provided", string(objectKey)),
					ResourcePath: propertyPath.String(),
				}
```